<a href="https://colab.research.google.com/github/pornnakanok/BasicProg_DataViz65/blob/main/Pandas101.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

## Meta Data = Data that describes data

## CSV = Comma Separated Values

## Pandas data frame

# นำ Data เข้า
- upload to colab
- link with google drive

## Upload data to colab

In [ ]:
data0 = pd.read_csv('/content/opendata_project.csv')

In [ ]:
data0.head()

row_number                project_id                             name_en  \
0      7880.0  5ee08d64dda791001299b2a6  Winning Residence Sukhumvit Praksa   
1     16689.0             project_11326                    Gift Condominium   
2       290.0  62739c4089233e001362b001    Sanrak 9 Sukprayoon Nong Krathum   
3     20719.0              project_2010                     Baan Sansabai 1   
4       199.0             project_16838          Baanpruksa Ratchapruek-345   

                              name_th propertytype_id propertytype_name_en  \
0  วินนิ่ง เรสซิเดนท์ สุขุมวิท แพรกษา               1       Detached House   
1                 กิ๊ฟท์ คอนโดมิเนียม               2                Condo   
2      แสนรัก 9 ศุขประยูร หนองกระทุ่ม               3             Townhome   
3                       บ้านแสนสบาย 1               1       Detached House   
4              บ้านพฤกษา ราชพฤกษ์-345               3             Townhome   

  propertytype_name_th  price_min     developer_id  \
0                 บ้าน  3790000.0  developer_10678   
1                คอนโด  1100000.0   developer_7319   
2             ทาวน์โฮม        0.0  developer_10510   
3                 บ้าน        0.0  developer_10510   
4             ทาวน์โฮม  1930000.0   developer_1026   

              developer_name_en  ... facility_meeting  facility_park  \
0  Winning Asset Plus Co., Ltd.  ...              NaN            NaN   
1      Gift Asset Land Co.,Ltd.  ...              NaN            1.0   
2                           NaN  ...              NaN            NaN   
3                           NaN  ...              NaN            NaN   
4  Preuksa Real Estate Co.,Ltd.  ...              NaN            1.0   

   facility_playground facility_pool facility_security         date_created  \
0                  NaN           1.0               1.0  2020-06-10 14:36:04   
1                  NaN           1.0               1.0  2017-10-30 15:38:48   
2                  NaN           NaN               NaN  2022-05-05 16:43:28   
3                  NaN           NaN               NaN  2016-09-05 15:32:15   
4                  NaN           1.0               1.0  2019-09-05 11:49:37   

           date_finish         date_updated  source  \
0                  NaN  2020-12-04 10:32:58  baania   
1  2013-01-01 07:00:00  2020-12-04 10:49:00  baania   
2                  NaN  2022-05-05 17:30:03  baania   
3  1970-01-02 07:00:00  2020-12-04 10:45:31  baania   
4  1970-01-02 07:00:00  2021-03-17 12:00:04  baania   

                                         url_project  
0  https://baania.com/th/project/วินนิ่ง-เรสซิเดน...  
1  https://baania.com/th/project/กิ๊ฟท์-คอนโดมิเน...  
2  https://baania.com/th/project/แสนรัก-9-ศุขประย...  
3  https://baania.com/th/project/บ้านแสนสบาย-1-pr...  
4  https://baania.com/th/project/บ้านพฤกษา-ราชพฤก...  

[5 rows x 45 columns]

In [ ]:
d = pd.read_csv('/content/opendata_unittype.csv')

In [ ]:
d.head()

row_number     unittype_id    project_id             name_th  \
0       27501   unittype_9565  project_2446           2 Bedroom   
1       28777  unittype_10267  project_3543  บ้านแฝดสองชั้น Z-2   
2       28865    unittype_471   project_143    1 Bedroom Type B   
3       24988  unittype_19506  project_7766    Studio Floor 1-2   
4       38596  unittype_10290  project_3551                TH 3   

   propertytype_id propertytype_name_en propertytype_name_th  \
0                2                Condo                คอนโด   
1            20000           Twin house              บ้านแฝด   
2                2                Condo                คอนโด   
3                2                Condo                คอนโด   
4                3             Townhome             ทาวน์โฮม   

   area_total_min_wa  area_usable_min  price_min  ...  count_room_guest  \
0                NaN             37.3  2500000.0  ...               1.0   
1               40.0            150.0  2390000.0  ...               NaN   
2                NaN             28.0        NaN  ...               NaN   
3                NaN             27.0   170000.0  ...               NaN   
4                NaN            182.0  2390000.0  ...               1.0   

   count_room_kitchen  count_room_living  count_room_maid  count_room_storage  \
0                 1.0                NaN              NaN                 NaN   
1                 NaN                NaN              NaN                 NaN   
2                 NaN                NaN              NaN                 NaN   
3                 NaN                NaN              NaN                 NaN   
4                 1.0                NaN              NaN                 2.0   

   count_room_utility         date_created         date_updated  source  \
0                 NaN  2021-11-19 18:00:09  2021-11-19 17:44:45  baania   
1                 NaN  2022-01-27 14:30:04  2022-01-27 14:22:55  baania   
2                 NaN  2022-01-14 15:00:10  2022-01-14 14:40:04  baania   
3                 NaN  2020-12-04 10:40:59  2020-10-14 01:06:48  baania   
4                 NaN  2021-11-12 13:30:08  2021-02-03 12:12:54  baania   

                                        url_unittype  
0  https://baania.com/th/project/เฟ-ลิค-คอนโด-ลาด...  
1  https://baania.com/th/project/เซน-บาย-ศิวาลัย-...  
2  https://baania.com/th/project/วันพลัส-นายน์ทีน...  
3  https://baania.com/th/project/บ้านสวนชมน้ำ-776...  
4  https://baania.com/th/project/เบล็สทาวน์-ศรีนค...  

[5 rows x 27 columns]

## link with google drive

data.go.th

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
data1 = pd.read_csv('/content/drive/MyDrive/BPDV65/opendata_unittype.csv')
data1.head()

row_number     unittype_id    project_id             name_th  \
0       27501   unittype_9565  project_2446           2 Bedroom   
1       28777  unittype_10267  project_3543  บ้านแฝดสองชั้น Z-2   
2       28865    unittype_471   project_143    1 Bedroom Type B   
3       24988  unittype_19506  project_7766    Studio Floor 1-2   
4       38596  unittype_10290  project_3551                TH 3   

   propertytype_id propertytype_name_en propertytype_name_th  \
0                2                Condo                คอนโด   
1            20000           Twin house              บ้านแฝด   
2                2                Condo                คอนโด   
3                2                Condo                คอนโด   
4                3             Townhome             ทาวน์โฮม   

   area_total_min_wa  area_usable_min  price_min  ...  count_room_guest  \
0                NaN             37.3  2500000.0  ...               1.0   
1               40.0            150.0  2390000.0  ...               NaN   
2                NaN             28.0        NaN  ...               NaN   
3                NaN             27.0   170000.0  ...               NaN   
4                NaN            182.0  2390000.0  ...               1.0   

   count_room_kitchen  count_room_living  count_room_maid  count_room_storage  \
0                 1.0                NaN              NaN                 NaN   
1                 NaN                NaN              NaN                 NaN   
2                 NaN                NaN              NaN                 NaN   
3                 NaN                NaN              NaN                 NaN   
4                 1.0                NaN              NaN                 2.0   

   count_room_utility         date_created         date_updated  source  \
0                 NaN  2021-11-19 18:00:09  2021-11-19 17:44:45  baania   
1                 NaN  2022-01-27 14:30:04  2022-01-27 14:22:55  baania   
2                 NaN  2022-01-14 15:00:10  2022-01-14 14:40:04  baania   
3                 NaN  2020-12-04 10:40:59  2020-10-14 01:06:48  baania   
4                 NaN  2021-11-12 13:30:08  2021-02-03 12:12:54  baania   

                                        url_unittype  
0  https://baania.com/th/project/เฟ-ลิค-คอนโด-ลาด...  
1  https://baania.com/th/project/เซน-บาย-ศิวาลัย-...  
2  https://baania.com/th/project/วันพลัส-นายน์ทีน...  
3  https://baania.com/th/project/บ้านสวนชมน้ำ-776...  
4  https://baania.com/th/project/เบล็สทาวน์-ศรีนค...  

[5 rows x 27 columns]

ถ้ารันแค่ data1 มันจะขึ้นข้อมูลทั้งหมด

In [ ]:
data1.head(10) #ถ้าไม่ใส่เลขในวงเล็บจะขึ้นแค่หัวตาราง 5 แถวแรก

row_number               unittype_id                project_id  \
0       27501             unittype_9565              project_2446   
1       28777            unittype_10267              project_3543   
2       28865              unittype_471               project_143   
3       24988            unittype_19506              project_7766   
4       38596            unittype_10290              project_3551   
5       30766             unittype_8026              project_2694   
6       33193            unittype_23233              project_7531   
7        1666            unittype_14501              project_5799   
8       41823  6215b4a7a441d30014e1f02f  6215b1cfc1620300140bc087   
9       34458            unittype_34564             project_14884   

                      name_th  propertytype_id propertytype_name_en  \
0                   2 Bedroom                2                Condo   
1          บ้านแฝดสองชั้น Z-2            20000           Twin house   
2            1 Bedroom Type B                2                Condo   
3            Studio Floor 1-2                2                Condo   
4                        TH 3                3             Townhome   
5               Luciano Grand                1       Detached House   
6         อาคารพาณิชย์สามชั้น                4           Commercial   
7    บ้านเดี่ยวสองชั้น Type A                1       Detached House   
8  บ้านเดี่ยวชั้นเดียว Type C                1       Detached House   
9         บ้านเดี่ยวชั้นเดียว                1       Detached House   

  propertytype_name_th  area_total_min_wa  area_usable_min   price_min  ...  \
0                คอนโด                NaN             37.3   2500000.0  ...   
1              บ้านแฝด               40.0            150.0   2390000.0  ...   
2                คอนโด                NaN             28.0         NaN  ...   
3                คอนโด                NaN             27.0    170000.0  ...   
4             ทาวน์โฮม                NaN            182.0   2390000.0  ...   
5                 บ้าน               70.5            198.0  11700000.0  ...   
6         อาคารพาณิชย์                NaN            180.0   2990000.0  ...   
7                 บ้าน                NaN              NaN   3300000.0  ...   
8                 บ้าน                NaN             80.0    990000.0  ...   
9                 บ้าน                NaN              NaN         NaN  ...   

   count_room_guest  count_room_kitchen  count_room_living  count_room_maid  \
0               1.0                 1.0                NaN              NaN   
1               NaN                 NaN                NaN              NaN   
2               NaN                 NaN                NaN              NaN   
3               NaN                 NaN                NaN              NaN   
4               1.0                 1.0                NaN              NaN   
5               NaN                 NaN                NaN              NaN   
6               NaN                 NaN                NaN              NaN   
7               NaN                 NaN                NaN              NaN   
8               NaN                 NaN                NaN              NaN   
9               NaN                 NaN                NaN              NaN   

   count_room_storage  count_room_utility         date_created  \
0                 NaN                 NaN  2021-11-19 18:00:09   
1                 NaN                 NaN  2022-01-27 14:30:04   
2                 NaN                 NaN  2022-01-14 15:00:10   
3                 NaN                 NaN  2020-12-04 10:40:59   
4                 2.0                 NaN  2021-11-12 13:30:08   
5                 NaN                 NaN  2021-11-12 13:22:43   
6                 NaN                 NaN  2020-12-04 10:46:42   
7                 NaN                 NaN  2020-12-04 10:36:59   
8                 NaN                 NaN  2022-02-23 12:00:12   
9                 NaN                 NaN  2020-12-04 10:45:1

In [ ]:
data1.tail(10) #head tail ดูข้อมูลในตารางคร่าวๆ

row_number               unittype_id                project_id  \
42615       36172            unittype_34140             project_14663   
42616       28981            unittype_25373             project_10133   
42617       35109             unittype_1459               project_711   
42618       22299            unittype_13674              project_5356   
42619       11944            unittype_34884             project_15005   
42620       37109            unittype_38508             project_16864   
42621          14            unittype_14016              project_5275   
42622       41009  617bde1687dd2300149d2195  617bdd47f62011001350de07   
42623        9333            unittype_37375             project_16389   
42624         780            unittype_17011              project_6762   

                   name_th  propertytype_id propertytype_name_en  \
42615            2 Bedroom                2                Condo   
42616    บ้านเดี่ยวสองชั้น                1       Detached House   
42617          บ้านแสงรุ่ง                3             Townhome   
42618  บ้านเดี่ยวชั้นเดียว                1       Detached House   
42619             Ametrine                1       Detached House   
42620               Canola                1       Detached House   
42621             Grandeur                1       Detached House   
42622               Grande                3             Townhome   
42623  บ้านเดี่ยวชั้นเดียว                1       Detached House   
42624             บ้านธารา                1       Detached House   

      propertytype_name_th  area_total_min_wa  area_usable_min   price_min  \
42615                คอนโด                NaN             60.0   3500000.0   
42616                 บ้าน                NaN            200.0         NaN   
42617             ทาวน์โฮม                NaN              NaN         NaN   
42618                 บ้าน                NaN            130.0  36500000.0   
42619                 บ้าน                NaN            353.0  23590000.0   
42620                 บ้าน                NaN            157.0         NaN   
42621                 บ้าน                NaN            223.0   7490000.0   
42622             ทาวน์โฮม                NaN            133.0         NaN   
42623                 บ้าน                NaN            180.0   3200000.0   
42624                 บ้าน                NaN            170.0   3790000.0   

       ...  count_room_guest  count_room_kitchen  count_room_living  \
42615  ...               1.0                 NaN                NaN   
42616  ...               1.0                 1.0                NaN   
42617  ...               NaN                 NaN                NaN   
42618  ...               NaN                 NaN                1.0   
42619  ...               1.0                 1.0                1.0   
42620  ...               NaN                 1.0                NaN   
42621  ...               NaN                 NaN                NaN   
42622  ...               1.0                 NaN                NaN   
42623  ...               NaN                 1.0                NaN   
42624  ...               1.0                 1.0                NaN   

       count_room_maid  count_room_storage  count_room_utility  \
42615              NaN                 NaN                 NaN   
42616              NaN                 1.0                 NaN   
42617              NaN                 NaN                 NaN   
42618              NaN                 NaN                 NaN   
42619              1.0                 1.0                 NaN   
42620              NaN                 NaN                 NaN   
42621              NaN                 NaN                 NaN   
42622              NaN                 NaN                 NaN   
42623              NaN                 NaN                 NaN   
42624              NaN                 NaN                 NaN   

              date_created         date_updated  source  \
42615  2022-02-28 15:30:05 

In [ ]:
data1.shape #(แถว,หลัก)

(42625, 27)

## Explore data

In [ ]:
data1.describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count  42625.000000     42625.000000        3747.000000     36999.000000   
mean   21313.000000       938.471859          44.413819       139.286943   
std    12304.921949      4220.823322          21.502884       111.939963   
min        1.000000         1.000000           1.000000        14.000000   
25%    10657.000000         1.000000          23.000000        60.000000   
50%    21313.000000         2.000000          50.000000       125.000000   
75%    31969.000000         3.000000          60.000000       179.000000   
max    42625.000000     20000.000000         335.000000      4350.000000   

          price_min  count_airconditioner  count_elevator  \
count  3.542800e+04           1780.000000             0.0   
mean   5.569699e+06              2.465730             NaN   
std    1.117088e+07              1.512717             NaN   
min    1.000000e+05              1.000000             NaN   
25%    1.950000e+06              1.000000             NaN   
50%    3.100000e+06              2.000000             NaN   
75%    5.290000e+06              3.000000             NaN   
max    5.800000e+08             10.000000             NaN   

       count_floor_internal  count_parking  count_room_bath  count_room_bed  \
count          29709.000000   23520.000000     41144.000000    40658.000000   
mean               1.945572       1.844090         2.286263        2.528137   
std                0.658217       0.759404         1.145901        1.158793   
min                0.000000       1.000000         0.000000        0.000000   
25%                2.000000       1.000000         1.000000        2.000000   
50%                2.000000       2.000000         2.000000        3.000000   
75%                2.000000       2.000000         3.000000        3.000000   
max                9.000000      10.000000        11.000000       10.000000   

       count_room_dinning  count_room_guest  count_room_kitchen  \
count        13670.000000      13044.000000        20454.000000   
mean             1.001902          1.010810            1.018138   
std              0.043572          0.106334            0.134549   
min              1.000000          1.000000            1.000000   
25%              1.000000          1.000000            1.000000   
50%              1.000000          1.000000            1.000000   
75%              1.000000          1.000000            1.000000   
max              2.000000          3.000000            3.000000   

       count_room_living  count_room_maid  count_room_storage  \
count       10026.000000      2068.000000         4108.000000   
mean            1.053860         1.048356            1.054771   
std             0.262145         0.227696            0.269673   
min             1.000000         1.000000            1.000000   
25%             1.000000         1.000000            1.000000   
50%             1.000000         1.000000            1.000000   
75%             1.000000         1.000000            1.000000   
max             5.000000         4.000000            5.000000   

       count_room_utility  
count          699.000000  
mean             1.058655  
std              0.284749  
min              1.000000  
25%              1.000000  
50%              1.000000  
75%              1.000000  
max              4.000000

- count จำนวนแถวทั้งหมด ไม่นับที่มีค่าว่าง
- area_total_min_wa (นับแค่ช่องที่ไม่ใช่ค่าว่าง) คือ พื้นที่ดินของตัวบ้าน ต่ำสุด (วา)
- mean
- std กระจายออกจากค่ากลาง พื้นที่เฉลี่ย 44 การกระจายตัวที่สุ่มมาจะเจอ?? งง ค่อยย้อนดู
- จากทั้งหมด 42625 หลัง มีบ้าน 25% ที่มีพื้นที่น้อยกว่า 23 ตารางวา
- จากทั้งหมด 42625 หลัง มีบ้าน 50% ที่มีพื้นที่น้อยกว่า 50 ตารางวา
- จากทั้งหมด 42625 หลัง มีบ้าน 75% ที่มีพื้นที่น้อยกว่า 60 ตารางวา
- บ้านที่มีพื้นที่เยอะสุดคือ 335 ตารางวา

NaN = Not a Number


## ชี้ค่าในตาราง (indexing)

In [ ]:
data1['propertytype_name_en']

0                 Condo
1            Twin house
2                 Condo
3                 Condo
4              Townhome
              ...      
42620    Detached House
42621    Detached House
42622          Townhome
42623    Detached House
42624    Detached House
Name: propertytype_name_en, Length: 42625, dtype: object

In [ ]:
data1['propertytype_name_en'][42619] #ชื่อคอลัมน์ ตามด้วยลำดับของข้อมูล

'Detached House'

.iloc

In [ ]:
data1.iloc[42619,5]

'Detached House'

In [ ]:
data1.iloc[-6,5]

'Detached House'

# Table Slicing

In [ ]:
data1.iloc[-20:,7:10]

area_total_min_wa  area_usable_min   price_min
42605                NaN              NaN   1300000.0
42606                NaN            349.0  18000000.0
42607                NaN             71.0  17000000.0
42608                NaN              NaN         NaN
42609                NaN              NaN         NaN
42610                NaN            250.0   5990000.0
42611                NaN              NaN         NaN
42612                NaN            154.0   4800000.0
42613               50.0            105.0   1900000.0
42614                NaN            142.1   4880000.0
42615                NaN             60.0   3500000.0
42616                NaN            200.0         NaN
42617                NaN              NaN         NaN
42618                NaN            130.0  36500000.0
42619                NaN            353.0  23590000.0
42620                NaN            157.0         NaN
42621                NaN            223.0   7490000.0
42622                NaN            133.0         NaN
42623                NaN            180.0   3200000.0
42624                NaN            170.0   3790000.0

## Table Query

In [ ]:
data1[data1['propertytype_name_en']=='Condo'] #ใช้ logical condition ในการเลือก data

row_number     unittype_id     project_id                      name_th  \
0           27501   unittype_9565   project_2446                    2 Bedroom   
2           28865    unittype_471    project_143             1 Bedroom Type B   
3           24988  unittype_19506   project_7766             Studio Floor 1-2   
10           5238  unittype_15162   project_4107             Studio Floor 5-8   
14          29911  unittype_25780  project_10313                    1 Bedroom   
...           ...             ...            ...                          ...   
42590       23548  unittype_15279   project_3976  1 Bedroom Tower 1 Floor 5-6   
42602       39384   unittype_3476    project_812             2 Bedroom Type A   
42604        1058   unittype_7949   project_2641            Typical 1 Bedroom   
42607       13546  unittype_40152  project_17505                    2 Bedroom   
42615       36172  unittype_34140  project_14663                    2 Bedroom   

       propertytype_id propertytype_name_en propertytype_name_th  \
0                    2                Condo                คอนโด   
2                    2                Condo                คอนโด   
3                    2                Condo                คอนโด   
10                   2                Condo                คอนโด   
14                   2                Condo                คอนโด   
...                ...                  ...                  ...   
42590                2                Condo                คอนโด   
42602                2                Condo                คอนโด   
42604                2                Condo                คอนโด   
42607                2                Condo                คอนโด   
42615                2                Condo                คอนโด   

       area_total_min_wa  area_usable_min   price_min  ...  count_room_guest  \
0                    NaN             37.3   2500000.0  ...               1.0   
2                    NaN             28.0         NaN  ...               NaN   
3                    NaN             27.0    170000.0  ...               NaN   
10                   NaN             30.0   1116000.0  ...               NaN   
14                   NaN             50.0   1700000.0  ...               NaN   
...                  ...              ...         ...  ...               ...   
42590                NaN             33.0    989000.0  ...               NaN   
42602                NaN             94.0   6345000.0  ...               NaN   
42604                NaN             34.0   3120000.0  ...               NaN   
42607                NaN             71.0  17000000.0  ...               NaN   
42615                NaN             60.0   3500000.0  ...               1.0   

       count_room_kitchen  count_room_living  count_room_maid  \
0                     1.0                NaN              NaN   
2                     NaN                NaN              NaN   
3                     NaN                NaN              NaN   
10                    1.0                NaN              NaN   
14                    1.0                NaN              NaN   
...                   ...                ...              ...   
42590                 NaN                NaN              NaN   
42602                 NaN                NaN              NaN   
42604                 1.0                1.0              NaN   
42607                 1.0                1.0              NaN   
42615                 NaN                NaN              NaN   

       count_room_storage  count_room_utility         date_created  \
0                     NaN                 NaN  2021-11-19 18:00:09   
2                     NaN                 NaN  2022-01-14 15:00:10   
3                     NaN                 NaN  2020-12-04 10:40:59   
10                    NaN                 NaN  2021-11-12 13:21:06   
14                    NaN                 NaN  2021-11-12 13:22:11   
...                   ...                 ...          

```pddf```[list_of_logical_values] คือการเลือก row ที่ logical values เป็น True

In [ ]:
data1['propertytype_name_en']=='Condo'

0         True
1        False
2         True
3         True
4        False
         ...  
42620    False
42621    False
42622    False
42623    False
42624    False
Name: propertytype_name_en, Length: 42625, dtype: bool

หาพื้นที่เฉลี่ยของบ้านที่ราคาเกิน 10 ล้าน

In [ ]:
data1[data1['price_min']> 10000000]

row_number               unittype_id                project_id  \
5           30766             unittype_8026              project_2694   
11           2452             unittype_8521              project_2906   
31          17879  5f682d5d0499a4001b30c305  5f6472855bdc5300198eb80a   
35           6309             unittype_3461               project_754   
37          26913            unittype_23137              project_9231   
...           ...                       ...                       ...   
42576       35213            unittype_22962              project_8609   
42606       24384            unittype_14126              project_5602   
42607       13546            unittype_40152             project_17505   
42618       22299            unittype_13674              project_5356   
42619       11944            unittype_34884             project_15005   

                   name_th  propertytype_id propertytype_name_en  \
5            Luciano Grand                1       Detached House   
11      New Villa Valdivia                1       Detached House   
31              2BR Duplex                2                Condo   
35                  Type C                1       Detached House   
37               4 Bedroom                2                Condo   
...                    ...              ...                  ...   
42576            Manhattan                1       Detached House   
42606      ทาวน์โฮมสามชั้น                3             Townhome   
42607            2 Bedroom                2                Condo   
42618  บ้านเดี่ยวชั้นเดียว                1       Detached House   
42619             Ametrine                1       Detached House   

      propertytype_name_th  area_total_min_wa  area_usable_min   price_min  \
5                     บ้าน               70.5            198.0  11700000.0   
11                    บ้าน                NaN            441.0  36900000.0   
31                   คอนโด                NaN            293.0  34222500.0   
35                    บ้าน                NaN            390.0  11000000.0   
37                   คอนโด                NaN            305.0  24000000.0   
...                    ...                ...              ...         ...   
42576                 บ้าน                NaN            465.0  28900000.0   
42606             ทาวน์โฮม                NaN            349.0  18000000.0   
42607                คอนโด                NaN             71.0  17000000.0   
42618                 บ้าน                NaN            130.0  36500000.0   
42619                 บ้าน                NaN            353.0  23590000.0   

       ...  count_room_guest  count_room_kitchen  count_room_living  \
5      ...               NaN                 NaN                NaN   
11     ...               1.0                 1.0                1.0   
31     ...               NaN                 1.0                1.0   
35     ...               NaN                 1.0                1.0   
37     ...               NaN                 NaN                NaN   
...    ...               ...                 ...                ...   
42576  ...               NaN                 2.0                2.0   
42606  ...               1.0                 2.0                NaN   
42607  ...               NaN                 1.0                1.0   
42618  ...               NaN                 NaN                1.0   
42619  ...               1.0                 1.0                1.0   

       count_room_maid  count_room_storage  count_room_utility  \
5                  NaN                 NaN                 NaN   
11                 1.0                 NaN                 NaN   
31                 NaN                 NaN                 NaN   
35                 NaN                 NaN                 NaN   
37                 NaN                 NaN                 NaN   
...                ...                 ...                 ...   
42576              1.0                 NaN                 Na

In [ ]:
data1[data1['price_min']> 10000000].describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count   3481.000000      3481.000000         304.000000      3354.000000   
mean   21066.205401       215.859523          52.622796       286.948606   
std    12478.117102      2035.944151          31.406177       185.081647   
min        1.000000         1.000000           1.000000        30.000000   
25%     9796.000000         1.000000          30.000000       155.000000   
50%    21525.000000         1.000000          51.400000       265.000000   
75%    32420.000000         2.000000          72.625000       362.750000   
max    42620.000000     20000.000000         335.000000      2548.000000   

          price_min  count_airconditioner  count_elevator  \
count  3.481000e+03            138.000000             0.0   
mean   2.577675e+07              4.536232             NaN   
std    2.789259e+07              2.015123             NaN   
min    1.003000e+07              1.000000             NaN   
25%    1.300000e+07              3.000000             NaN   
50%    1.700000e+07              4.000000             NaN   
75%    2.790200e+07              5.750000             NaN   
max    5.800000e+08             10.000000             NaN   

       count_floor_internal  count_parking  count_room_bath  count_room_bed  \
count           2149.000000    1784.000000      3420.000000     3411.000000   
mean               2.346208       2.867152         3.687427        3.357666   
std                0.816617       1.284865         1.458689        1.101538   
min                0.000000       1.000000         0.000000        0.000000   
25%                2.000000       2.000000         3.000000        3.000000   
50%                2.000000       3.000000         4.000000        3.000000   
75%                3.000000       3.000000         5.000000        4.000000   
max                7.000000      10.000000        10.000000       10.000000   

       count_room_dinning  count_room_guest  count_room_kitchen  \
count         1570.000000       1236.000000         1897.000000   
mean             1.007006          1.046926            1.061676   
std              0.083437          0.222751            0.247118   
min              1.000000          1.000000            1.000000   
25%              1.000000          1.000000            1.000000   
50%              1.000000          1.000000            1.000000   
75%              1.000000          1.000000            1.000000   
max              2.000000          3.000000            3.000000   

       count_room_living  count_room_maid  count_room_storage  \
count        1396.000000       989.000000          497.000000   
mean            1.169054         1.070779            1.108652   
std             0.473001         0.275603            0.411853   
min             1.000000         1.000000            1.000000   
25%             1.000000         1.000000            1.000000   
50%             1.000000         1.000000            1.000000   
75%             1.000000         1.000000            1.000000   
max             5.000000         4.000000            4.000000   

       count_room_utility  
count           83.000000  
mean             1.120482  
std              0.424769  
min              1.000000  
25%              1.000000  
50%              1.000000  
75%              1.000000  
max              4.000000

# Quiz 4

1. ราคาเฉลี่ยของบ้านที่มีราคาระหว่าง 3-5 ล้านบาท 
2. พื้นที่ และ จำนวนห้องนอนของบ้านที่แพงที่สุด
3. จำนวนที่จอดรถและห้องน้ำของบ้านที่ถูกที่สุด
4. ชนิดของบ้านที่แพงที่สุด
5. บ้านชนิดไหนมีปริมาณมากที่สุด

1. ราคาเฉลี่ยของบ้านที่มีราคาระหว่าง 3-5 ล้านบาท

ตอบ 3.928111 ล้านบาท

In [ ]:
(data1['price_min'] > 3000000) & (data1['price_min']< 5000000)

0        False
1        False
2        False
3        False
4        False
         ...  
42620    False
42621    False
42622    False
42623     True
42624     True
Name: price_min, Length: 42625, dtype: bool

In [ ]:
data1[(data1['price_min'] > 3000000) & (data1['price_min']< 5000000)].describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count   8433.000000      8433.000000        1011.000000      7723.000000   
mean   21117.962528      1291.864698          46.622888       138.730280   
std    12334.267520      4912.977037          18.512287        66.391354   
min        2.000000         1.000000           2.000000        20.100000   
25%    10015.000000         1.000000          29.650000        97.000000   
50%    21818.000000         2.000000          50.000000       145.000000   
75%    32145.000000         3.000000          60.000000       178.000000   
max    42617.000000     20000.000000          99.000000       896.000000   

          price_min  count_airconditioner  count_elevator  \
count  8.433000e+03            387.000000             0.0   
mean   3.928111e+06              2.666667             NaN   
std    5.507601e+05              1.128987             NaN   
min    3.002000e+06              1.000000             NaN   
25%    3.490000e+06              2.000000             NaN   
50%    3.890000e+06              3.000000             NaN   
75%    4.390000e+06              3.000000             NaN   
max    4.999000e+06              7.000000             NaN   

       count_floor_internal  count_parking  count_room_bath  count_room_bed  \
count           6526.000000    5305.000000      8247.000000     8120.000000   
mean               2.136531       1.883318         2.414454        2.739901   
std                0.575003       0.494459         0.913251        0.983963   
min                0.000000       1.000000         0.000000        0.000000   
25%                2.000000       2.000000         2.000000        2.000000   
50%                2.000000       2.000000         3.000000        3.000000   
75%                2.000000       2.000000         3.000000        3.000000   
max                4.000000      10.000000         9.000000       10.000000   

       count_room_dinning  count_room_guest  count_room_kitchen  \
count         3131.000000       3068.000000         4531.000000   
mean             1.001278          1.005541            1.016553   
std              0.035726          0.074244            0.127602   
min              1.000000          1.000000            1.000000   
25%              1.000000          1.000000            1.000000   
50%              1.000000          1.000000            1.000000   
75%              1.000000          1.000000            1.000000   
max              2.000000          2.000000            2.000000   

       count_room_living  count_room_maid  count_room_storage  \
count        1871.000000       138.000000         1172.000000   
mean            1.028327         1.036232            1.038396   
std             0.172274         0.223098            0.196624   
min             1.000000         1.000000            1.000000   
25%             1.000000         1.000000            1.000000   
50%             1.000000         1.000000            1.000000   
75%             1.000000         1.000000            1.000000   
max             3.000000         3.000000            3.000000   

       count_room_utility  
count          157.000000  
mean             1.108280  
std              0.401593  
min              1.000000  
25%              1.000000  
50%              1.000000  
75%              1.000000  
max              3.000000

In [18]:
price_mean = data1[(data1['price_min'] > 3000000) & (data1['price_min']< 5000000)]['price_min'].mean()

In [21]:
print(f'ราคาเฉลี่ยของบ้านที่มีราคาระหว่าง 3-5 ล้านบาท คือ {int(price_mean):,}')

ราคาเฉลี่ยของบ้านที่มีราคาระหว่าง 3-5 ล้านบาท คือ 3,928,110


2. พื้นที่ และ จำนวนห้องนอนของบ้านที่แพงที่สุด

area_total_min_wa และ count_room_bed

In [22]:
data1['price_min'].max()

580000000.0

In [23]:
data1[data1['price_min']== data1['price_min'].max()]

row_number     unittype_id     project_id    name_th  propertytype_id  \
33122        8629  unittype_37324  project_16305  Penthouse                2   

      propertytype_name_en propertytype_name_th  area_total_min_wa  \
33122                Condo                คอนโด                NaN   

       area_usable_min    price_min  ...  count_room_guest  \
33122           1182.0  580000000.0  ...               NaN   

       count_room_kitchen  count_room_living  count_room_maid  \
33122                 NaN                NaN              NaN   

       count_room_storage  count_room_utility         date_created  \
33122                 NaN                 NaN  2021-11-12 13:20:31   

              date_updated  source  \
33122  2020-10-14 00:55:50  baania   

                                            url_unittype  
33122  https://baania.com/th/project/เดอะ-สุโขทัย-เรส...  

[1 rows x 27 columns]

In [24]:
data1[data1['price_min']== data1['price_min'].max()][['propertytype_name_en','area_usable_min','count_room_bed','price_min' ]]

propertytype_name_en  area_usable_min  count_room_bed    price_min
33122                Condo           1182.0             3.0  580000000.0

In [12]:
data1[data1['price_min']== 5.800000e+08]['area_usable_min']

33122    1182.0
Name: area_usable_min, dtype: float64

In [11]:
data1[data1['price_min']== 5.800000e+08]['count_room_bed']

33122    3.0
Name: count_room_bed, dtype: float64

In [13]:
data1[data1['price_min']== 5.800000e+08][['area_usable_min','count_room_bed']]

area_usable_min  count_room_bed
33122           1182.0             3.0

In [5]:
data1[data1['price_min']== 5.800000e+08].describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count         1.0              1.0                0.0              1.0   
mean       8629.0              2.0                NaN           1182.0   
std           NaN              NaN                NaN              NaN   
min        8629.0              2.0                NaN           1182.0   
25%        8629.0              2.0                NaN           1182.0   
50%        8629.0              2.0                NaN           1182.0   
75%        8629.0              2.0                NaN           1182.0   
max        8629.0              2.0                NaN           1182.0   

         price_min  count_airconditioner  count_elevator  \
count          1.0                   0.0             0.0   
mean   580000000.0                   NaN             NaN   
std            NaN                   NaN             NaN   
min    580000000.0                   NaN             NaN   
25%    580000000.0                   NaN             NaN   
50%    580000000.0                   NaN             NaN   
75%    580000000.0                   NaN             NaN   
max    580000000.0                   NaN             NaN   

       count_floor_internal  count_parking  count_room_bath  count_room_bed  \
count                   0.0            0.0              1.0             1.0   
mean                    NaN            NaN              5.0             3.0   
std                     NaN            NaN              NaN             NaN   
min                     NaN            NaN              5.0             3.0   
25%                     NaN            NaN              5.0             3.0   
50%                     NaN            NaN              5.0             3.0   
75%                     NaN            NaN              5.0             3.0   
max                     NaN            NaN              5.0             3.0   

       count_room_dinning  count_room_guest  count_room_kitchen  \
count                 0.0               0.0                 0.0   
mean                  NaN               NaN                 NaN   
std                   NaN               NaN                 NaN   
min                   NaN               NaN                 NaN   
25%                   NaN               NaN                 NaN   
50%                   NaN               NaN                 NaN   
75%                   NaN               NaN                 NaN   
max                   NaN               NaN                 NaN   

       count_room_living  count_room_maid  count_room_storage  \
count                0.0              0.0                 0.0   
mean                 NaN              NaN                 NaN   
std                  NaN              NaN                 NaN   
min                  NaN              NaN                 NaN   
25%                  NaN              NaN                 NaN   
50%                  NaN              NaN                 NaN   
75%                  NaN              NaN                 NaN   
max                  NaN              NaN                 NaN   

       count_room_utility  
count                 0.0  
mean                  NaN  
std                   NaN  
min                   NaN  
25%                   NaN  
50%                   NaN  
75%                   NaN  
max                   NaN

3. จำนวนที่จอดรถและห้องน้ำของบ้านที่ถูกที่สุด

count_parking, count_room_bath 

In [6]:
data1.describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count  42625.000000     42625.000000        3747.000000     36999.000000   
mean   21313.000000       938.471859          44.413819       139.286943   
std    12304.921949      4220.823322          21.502884       111.939963   
min        1.000000         1.000000           1.000000        14.000000   
25%    10657.000000         1.000000          23.000000        60.000000   
50%    21313.000000         2.000000          50.000000       125.000000   
75%    31969.000000         3.000000          60.000000       179.000000   
max    42625.000000     20000.000000         335.000000      4350.000000   

          price_min  count_airconditioner  count_elevator  \
count  3.542800e+04           1780.000000             0.0   
mean   5.569699e+06              2.465730             NaN   
std    1.117088e+07              1.512717             NaN   
min    1.000000e+05              1.000000             NaN   
25%    1.950000e+06              1.000000             NaN   
50%    3.100000e+06              2.000000             NaN   
75%    5.290000e+06              3.000000             NaN   
max    5.800000e+08             10.000000             NaN   

       count_floor_internal  count_parking  count_room_bath  count_room_bed  \
count          29709.000000   23520.000000     41144.000000    40658.000000   
mean               1.945572       1.844090         2.286263        2.528137   
std                0.658217       0.759404         1.145901        1.158793   
min                0.000000       1.000000         0.000000        0.000000   
25%                2.000000       1.000000         1.000000        2.000000   
50%                2.000000       2.000000         2.000000        3.000000   
75%                2.000000       2.000000         3.000000        3.000000   
max                9.000000      10.000000        11.000000       10.000000   

       count_room_dinning  count_room_guest  count_room_kitchen  \
count        13670.000000      13044.000000        20454.000000   
mean             1.001902          1.010810            1.018138   
std              0.043572          0.106334            0.134549   
min              1.000000          1.000000            1.000000   
25%              1.000000          1.000000            1.000000   
50%              1.000000          1.000000            1.000000   
75%              1.000000          1.000000            1.000000   
max              2.000000          3.000000            3.000000   

       count_room_living  count_room_maid  count_room_storage  \
count       10026.000000      2068.000000         4108.000000   
mean            1.053860         1.048356            1.054771   
std             0.262145         0.227696            0.269673   
min             1.000000         1.000000            1.000000   
25%             1.000000         1.000000            1.000000   
50%             1.000000         1.000000            1.000000   
75%             1.000000         1.000000            1.000000   
max             5.000000         4.000000            5.000000   

       count_room_utility  
count          699.000000  
mean             1.058655  
std              0.284749  
min              1.000000  
25%              1.000000  
50%              1.000000  
75%              1.000000  
max              4.000000

In [8]:
data1[data1['price_min']== 1.000000e+05]['count_parking']

8640    NaN
11019   NaN
17941   NaN
30364   NaN
Name: count_parking, dtype: float64

In [9]:
data1[data1['price_min']== 1.000000e+05]['count_room_bath']

8640     1.0
11019    1.0
17941    1.0
30364    1.0
Name: count_room_bath, dtype: float64

In [15]:
data1[data1['price_min']== 1.000000e+05][['count_parking','count_room_bath']]

count_parking  count_room_bath
8640             NaN              1.0
11019            NaN              1.0
17941            NaN              1.0
30364            NaN              1.0

In [7]:
data1[data1['price_min']== 1.000000e+05].describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count      4.000000              4.0                0.0         4.000000   
mean   14920.000000              2.0                NaN        30.000000   
std    16480.011893              0.0                NaN         4.242641   
min      490.000000              2.0                NaN        24.000000   
25%      728.500000              2.0                NaN        28.500000   
50%    14902.000000              2.0                NaN        31.500000   
75%    29093.500000              2.0                NaN        33.000000   
max    29386.000000              2.0                NaN        33.000000   

       price_min  count_airconditioner  count_elevator  count_floor_internal  \
count        4.0                   0.0             0.0                   0.0   
mean    100000.0                   NaN             NaN                   NaN   
std          0.0                   NaN             NaN                   NaN   
min     100000.0                   NaN             NaN                   NaN   
25%     100000.0                   NaN             NaN                   NaN   
50%     100000.0                   NaN             NaN                   NaN   
75%     100000.0                   NaN             NaN                   NaN   
max     100000.0                   NaN             NaN                   NaN   

       count_parking  count_room_bath  count_room_bed  count_room_dinning  \
count            0.0              4.0            4.00                 0.0   
mean             NaN              1.0            0.75                 NaN   
std              NaN              0.0            0.50                 NaN   
min              NaN              1.0            0.00                 NaN   
25%              NaN              1.0            0.75                 NaN   
50%              NaN              1.0            1.00                 NaN   
75%              NaN              1.0            1.00                 NaN   
max              NaN              1.0            1.00                 NaN   

       count_room_guest  count_room_kitchen  count_room_living  \
count               1.0                 1.0                0.0   
mean                1.0                 1.0                NaN   
std                 NaN                 NaN                NaN   
min                 1.0                 1.0                NaN   
25%                 1.0                 1.0                NaN   
50%                 1.0                 1.0                NaN   
75%                 1.0                 1.0                NaN   
max                 1.0                 1.0                NaN   

       count_room_maid  count_room_storage  count_room_utility  
count              0.0                 0.0                 0.0  
mean               NaN                 NaN                 NaN  
std                NaN                 NaN                 NaN  
min                NaN                 NaN                 NaN  
25%                NaN                 NaN                 NaN  
50%                NaN                 NaN                 NaN  
75%                NaN                 NaN                 NaN  
max                NaN                 NaN                 NaN

In [25]:
data1[data1['price_min']== data1['price_min'].min()][['propertytype_name_en','count_parking','count_room_bath','price_min' ]]

propertytype_name_en  count_parking  count_room_bath  price_min
8640                 Condo            NaN              1.0   100000.0
11019                Condo            NaN              1.0   100000.0
17941                Condo            NaN              1.0   100000.0
30364                Condo            NaN              1.0   100000.0

4. ชนิดของบ้านที่แพงที่สุด

ตอบ Condo

In [ ]:
data1['propertytype_name_en'][data1['price_min']== 5.800000e+08]

33122    Condo
Name: propertytype_name_en, dtype: object

5. บ้านชนิดไหนมีปริมาณมากที่สุด

ตอบ Detached House

In [ ]:
data1['propertytype_name_en'].describe()

count              42625
unique                11
top       Detached House
freq               17670
Name: propertytype_name_en, dtype: object

unique = ชนิดของบ้านที่ต่างกัน

top = มีเยอะสุด

In [ ]:
data1[data1['propertytype_name_en']=='Detached House'].describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count  17670.000000          17670.0        2436.000000     14652.000000   
mean   21332.680872              1.0          55.199080       199.252515   
std    12276.498246              0.0          18.355487       117.514606   
min        1.000000              1.0           1.000000        14.000000   
25%    10801.250000              1.0          50.000000       136.000000   
50%    20456.500000              1.0          55.000000       171.000000   
75%    32363.750000              1.0          63.400000       225.000000   
max    42625.000000              1.0         335.000000      3175.000000   

          price_min  count_airconditioner  count_elevator  \
count  1.397600e+04            806.000000             0.0   
mean   6.674175e+06              2.971464             NaN   
std    1.066470e+07              1.727506             NaN   
min    1.500000e+05              1.000000             NaN   
25%    2.690000e+06              2.000000             NaN   
50%    4.000000e+06              3.000000             NaN   
75%    6.500000e+06              4.000000             NaN   
max    2.850000e+08             10.000000             NaN   

       count_floor_internal  count_parking  count_room_bath  count_room_bed  \
count          17418.000000   14742.000000     17012.000000    16896.000000   
mean               1.748077       1.990503         2.829650        3.216501   
std                0.507313       0.751123         1.079692        0.730838   
min                0.000000       1.000000         0.000000        0.000000   
25%                1.000000       2.000000         2.000000        3.000000   
50%                2.000000       2.000000         3.000000        3.000000   
75%                2.000000       2.000000         3.000000        4.000000   
max                7.000000      10.000000        10.000000       10.000000   

       count_room_dinning  count_room_guest  count_room_kitchen  \
count         6916.000000       7859.000000        10094.000000   
mean             1.002169          1.013997            1.028730   
std              0.046524          0.118562            0.168237   
min              1.000000          1.000000            1.000000   
25%              1.000000          1.000000            1.000000   
50%              1.000000          1.000000            1.000000   
75%              1.000000          1.000000            1.000000   
max              2.000000          3.000000            3.000000   

       count_room_living  count_room_maid  count_room_storage  \
count        3945.000000      1849.000000         2818.000000   
mean            1.110520         1.050297            1.049326   
std             0.367207         0.232996            0.251473   
min             1.000000         1.000000            1.000000   
25%             1.000000         1.000000            1.000000   
50%             1.000000         1.000000            1.000000   
75%             1.000000         1.000000            1.000000   
max             5.000000         4.000000            4.000000   

       count_room_utility  
count          259.000000  
mean             1.030888  
std              0.173349  
min              1.000000  
25%              1.000000  
50%              1.000000  
75%              1.000000  
max              2.000000

$.isnull()$ ใช้ตรวจสอบว่าเป็น NaN หรือไม่

In [ ]:
import numpy as np

In [ ]:
import os

In [ ]:
path = '/content/drive/My Drive/dataviz_2021_data'

In [ ]:
covid_file_path = os.path.join(path,'pm-20-jan-2021.csv') #บอกโลเคชั่นของไฟล์
print(covid_file_path)

In [ ]:
a = path + '/pm-18-jan-2021.csv' 
print(a)

## load data to memory

In [ ]:
data_covid = pd.read_csv(covid_file_path)
data_covid